src/axom/primal/operators/detail/intersect_patch_impl.hpp

```diff
@@ -0,0 +1,248 @@
+                          axom::Array<T> &tp,
+                          axom::Array<T> &up,
+                          axom::Array<T> &vp,
+                          int order_u,
+                          int order_v,
+                          double u_offset,
+                          double u_scale,
+                          double v_offset,
+                          double v_scale,
+                          double sq_tol,
+                          double EPS,
+                          bool isRay);
+
+//------------------------------ IMPLEMENTATIONS ------------------------------
+
+template <typename T>
+bool intersect_line_patch(const Line<T, 3> &line,
+                          const BezierPatch<T, 3> &patch,
+                          axom::Array<T> &tp,
+                          axom::Array<T> &up,
+                          axom::Array<T> &vp,
+                          int order_u,
+                          int order_v,
+                          double u_offset,
+                          double u_scale,
+                          double v_offset,
+                          double v_scale,
+                          double sq_tol,
+                          double EPS,
+                          bool isRay)
+{
+  using BPatch = BezierPatch<T, 3>;
+
+  // Check bounding box to short-circuit the intersection
+  //  Need to expand the box a bit so that intersections near subdivision boundaries
+  //  are accurately recorded
+  Point<T, 3> ip;
+  if(!intersect(line, patch.boundingBox().scale(1.5), ip))
+  {
+    return false;
+  }
+
+  bool foundIntersection = false;
+  if(patch.isBilinear(sq_tol))
+  {
+    // Store candidate intersection points
+    axom::Array<T> tc, uc, vc;
+
+    foundIntersection =
+      detail::intersect_line_bilinear_patch(line,
+                                            patch(0, 0),
+                                            patch(order_u, 0),
+                                            patch(order_u, order_v),
+                                            patch(0, order_v),
+                                            tc,
+                                            uc,
+                                            vc,
+                                            EPS,
+                                            isRay);
+
+    if(!foundIntersection)
+    {
+      return false;
+    }
+
+    foundIntersection = false;
+    for(int i = 0; i < tc.size(); ++i)
+    {
+      const T t0 = tc[i];
+      const T u0 = uc[i];
+      const T v0 = vc[i];
+
+      // Use EPS to record points near the boundary of the bilinear approximation
+      if(u0 >= -EPS / u_scale && u0 <= 1.0 + EPS / u_scale &&
+         v0 >= -EPS / v_scale && v0 <= 1.0 + EPS / v_scale)
+      {
+        if(t0 >= -EPS || !isRay)
+        {
+          up.push_back(u_offset + u0 * u_scale);
+          vp.push_back(v_offset + v0 * v_scale);
+          tp.push_back(t0);
+
+          foundIntersection = true;
```

Comment:
Ah, this is outdated behavior from a different way I had of handling half-open intervals. We want to check that the parameters of the intersection are valid *outside* the original `intersect_line_bilinear_patch` function, since the definition of "valid" depends on the scale in each direction. I've fixed it to be a little more sensible on a first glance.